bug: Rig MCP tools does not work with rmcp 0.6.1

- [x] I have looked for existing issues (including closed) about this

## Bug Report
I'm writing an app which uses rmcp 0.6.1, and when trying to write an agent, using rig-core 0.19, which locks to rmcp 0.5.0, I get the following error: 

```
cargo build               
    Blocking waiting for file lock on build directory
   Compiling rmcp-macros v0.5.0
   Compiling rmcp-macros v0.6.1
   Compiling rmcp v0.5.0
   Compiling rmcp v0.6.1
   Compiling rig-core v0.19.0
   Compiling lakenuc v0.1.0 (pathto/cli)
error[E0308]: arguments to this method are incorrect
   --> src/agent.rs:91:37
    |
91  | ...                   builder.rmcp_tool(tool, tc.client.peer().clone())
    |                               ^^^^^^^^^ ---- expected `rmcp::model::tool::Tool`, found `rmcp::model::Tool`
    |
note: two different versions of crate `rmcp` are being used; two types coming from two different versions of the same crate are different types even if they look the same
   --> pathto.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rmcp-0.6.1/src/model/tool.rs:15:1
    |
15  | pub struct Tool {
    | ^^^^^^^^^^^^^^^ this is the found type `rmcp::model::Tool`
    |
   ::: pathto.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rmcp-0.5.0/src/model/tool.rs:15:1
    |
15  | pub struct Tool {
    | ^^^^^^^^^^^^^^^ this is the expected type `rmcp::model::tool::Tool`
    |
   ::: src/agent.rs:3:5
    |
3   | use rig::{client::CompletionClient, completion::Prompt, providers::ollama};
    |     --- one version of crate `rmcp` used here, as a dependency of crate `rig`
    |
   ::: src/tool_client.rs:3:5
    |
3   | use rmcp::{
    |     ---- one version of crate `rmcp` used here, as a direct dependency of the current crate
    = help: you can use `cargo tree` to explore your dependency tree
note: expected `rmcp::service::Peer<rmcp::service::client::RoleClient>`, found `Peer<RoleClient>`
   --> src/agent.rs:91:53
    |
91  | ...                   builder.rmcp_tool(tool, tc.client.peer().clone())
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^
note: two different versions of crate `rmcp` are being used; two types coming from two different versions of the same crate are different types even if they look the same
   --> pathto.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rmcp-0.6.1/src/service.rs:313:1
    |
313 | pub struct Peer<R: ServiceRole> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ this is the found type `Peer`
    |
   ::: pathto.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rmcp-0.5.0/src/service.rs:313:1
    |
313 | pub struct Peer<R: ServiceRole> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ this is the expected type `rmcp::service::Peer`
    |
   ::: src/agent.rs:3:5
    |
3   | use rig::{client::CompletionClient, completion::Prompt, providers::ollama};
    |     --- one version of crate `rmcp` used here, as a dependency of crate `rig`
    |
   ::: src/tool_client.rs:3:5
    |
3   | use rmcp::{
    |     ---- one version of crate `rmcp` used here, as a direct dependency of the current crate
    = help: you can use `cargo tree` to explore your dependency tree
note: method defined here
   --> pathto.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rig-core-0.19.0/src/agent/builder.rs:127:12
    |
127 |     pub fn rmcp_tool(mut self, tool: rmcp::model::Tool, client: rmcp::service::ServerSink) -> Self {
    |            ^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `lakenuc` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

## Reproduction

Simply running https://github.com/0xPlaygrounds/rig/blob/2013a6c27a2da0ae425670de2f65c1b296d012d1/rig-core/examples/rmcp.rs#L267 with rmcp 0.6.1 in the environment throws the same bug.

### main.rs
```
//! An example of how you can use `rmcp` with Rig to create an MCP friendly agent.
use std::sync::Arc;

use anyhow::Context;
use rmcp::{transport::{ConfigureCommandExt, TokioChildProcess}, ServiceExt};

use rig::{
    client::{CompletionClient, ProviderClient},
    completion::Prompt,
    providers::openai,
};
use rmcp::{
    model::{CallToolRequestParam, ClientCapabilities, ClientInfo, Implementation, InitializeRequestParam, Tool}, service::{RoleClient, RunningService}
};
use rmcp::{
    RoleServer, ServerHandler,
    handler::server::{router::tool::ToolRouter},
    model::*,
    schemars,
    service::RequestContext,
    tool, tool_handler, tool_router,
};
use serde_json::json;
use tokio::sync::Mutex;

use hyper_util::{
    rt::{TokioExecutor, TokioIo},
    server::conn::auto::Builder,
};

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    tracing_subscriber::fmt::init();

    // Create transport to start the MCP server as a child process
    let transport = TokioChildProcess::new(
        tokio::process::Command::new("server").configure(|cmd| {
            cmd.kill_on_drop(true);
        }),
    ).context("Failed to create transport to MCP server")?;

    let client_info = ClientInfo {
        protocol_version: Default::default(),
        capabilities: ClientCapabilities::default(),
        client_info: Implementation {
            name: "rig-core".to_string(),
            version: "0.13.0".to_string(),
        },
    };

    let client = client_info.serve(transport).await.inspect_err(|e| {
        tracing::error!("client error: {:?}", e);
    })?;

    // Initialize
    let server_info = client.peer_info();
    tracing::info!("Connected to server: {server_info:#?}");

    // List tools
    let tools: Vec<Tool> = client.list_tools(Default::default()).await?.tools;

    // takes the `OPENAI_API_KEY` as an env var on usage
    let openai_client = openai::Client::from_env();
    let agent = openai_client
        .agent("gpt-4o")
        .preamble("You are a helpful assistant who has access to a number of tools from an MCP server designed to be used for incrementing and decrementing a counter.");

    let agent = tools
        .into_iter()
        .fold(agent, |agent, tool| agent.rmcp_tool(tool, client.clone()))
        .build();

    let res = agent.prompt("What is 2+5?").multi_turn(2).await.unwrap();

    println!("GPT-4o: {res}");

    Ok(())
}
```

### cargo.toml
```
[package]
name = "test-rig"
version = "0.1.0"
edition = "2024"

[dependencies]
rig-core = { version = "0.19.0", features = ["rmcp"] }
tokio = { version = "1.47", features = ["macros", "rt-multi-thread", "process"] }
rmcp = { version = "0.6.1", features = ["transport-child-process", "client"] }
anyhow = "1.0"
serde_json = "1.0"
serde = { version = "1.0.219", features = ["derive"] }
clap = { version = "4.5", features = ["derive"] }
# dependencies for better UX
colored = "3.0"
log = "0.4.27"
env_logger = "0.11.8"
tracing = "0.1"
tracing-subscriber = "0.3"
hyper-util = "0.1"
```

## Expected behavior
Rig could work with rmcp >=0.6.1.

--- a/rig-core/Cargo.toml
+++ b/rig-core/Cargo.toml
@@ -57,8 +57,10 @@ base64 = { workspace = true }
 hyper-util = { version = "0.1.14", features = ["service", "server"] }
 rmcp = { version = "0.6", features = [
     "client",
+    "macros",
     "reqwest",                                  # required for some strange reason
     "transport-streamable-http-client",
+    "transport-streamable-http-client-reqwest",
     "transport-streamable-http-server-session",
     "transport-streamable-http-server",
     "transport-worker",

--- a/rig-core/Cargo.toml
+++ b/rig-core/Cargo.toml
@@ -40,7 +40,7 @@ thiserror = { workspace = true }
 tracing = { workspace = true }
 url = { workspace = true }
 worker = { workspace = true, optional = true }
-rmcp = { version = "0.5", optional = true, features = ["client"] }
+rmcp = { version = "0.6", optional = true, features = ["client"] }
 reqwest-eventsource = { workspace = true }
 tokio = { workspace = true, features = ["sync"] }
 
@@ -55,7 +55,7 @@ base64 = { workspace = true }
 
 # Required for `rmcp` example
 hyper-util = { version = "0.1.14", features = ["service", "server"] }
-rmcp = { version = "0.5", features = [
+rmcp = { version = "0.6", features = [
     "client",
     "reqwest",                                  # required for some strange reason
     "transport-streamable-http-client",

--- a/rig-core/examples/rmcp.rs
+++ b/rig-core/examples/rmcp.rs
@@ -10,7 +10,7 @@ use rig::{
 };
 use rmcp::{
     RoleServer, ServerHandler,
-    handler::server::{router::tool::ToolRouter, tool::Parameters},
+    handler::server::{router::tool::ToolRouter, wrapper::Parameters},
     model::*,
     schemars,
     service::RequestContext,

--- a/rig-core/src/tool.rs
+++ b/rig-core/src/tool.rs
@@ -280,18 +280,22 @@ pub mod rmcp {
                 if let Some(true) = result.is_error {
                     let error_msg = result
                         .content
-                        .as_deref()
-                        .and_then(|errors| errors.first())
-                        .and_then(|error| error.as_text())
-                        .map(|raw| raw.text.as_str())
-                        .unwrap_or("No error message returned");
-                    return Err(McpToolError(error_msg.to_string()).into());
+                        .into_iter()
+                        .map(|x| x.raw.as_text().map(|y| y.to_owned()))
+                        .map(|x| x.map(|x| x.clone().text))
+                        .collect::<Option<Vec<String>>>();
+
+                    let error_message = error_msg.map(|x| x.join("\n"));
+                    if let Some(error_message) = error_message {
+                        return Err(McpToolError(error_message).into());
+                    } else {
+                        return Err(McpToolError("No message returned".to_string()).into());
+                    }
                 };
 
                 Ok(result
                     .content
                     .into_iter()
-                    .flatten()
                     .map(|c| match c.raw {
                         rmcp::model::RawContent::Text(raw) => raw.text,
                         rmcp::model::RawContent::Image(raw) => {
@@ -302,6 +306,7 @@ pub mod rmcp {
                                 uri,
                                 mime_type,
                                 text,
+                                ..
                             } => {
                                 format!(
                                     "{mime_type}{uri}:{text}",
@@ -314,6 +319,7 @@ pub mod rmcp {
                                 uri,
                                 mime_type,
                                 blob,
+                                ..
                             } => format!(
                                 "{mime_type}{uri}:{blob}",
                                 mime_type = mime_type
@@ -323,6 +329,9 @@ pub mod rmcp {
                         },
                         RawContent::Audio(_) => {
                             unimplemented!("Support for audio results from an MCP tool is currently unimplemented. Come back later!")
+                        }
+                        thing => {
+                            unimplemented!("Unsupported type found: {thing:?}")
                         }
                     })
                     .collect::<String>())

